### PR TITLE
feature/sentry.io for web

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Install sentry.io for web app

**Environment**: Staging(main branch)

## Type of change

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: apps/web/next.config.js [hunk 1]
```diff
@@ -228,6 +229,18 @@ const nextConfig = {
 
     return redirects;
   },
+  sentry: {
+    hideSourceMaps: true,
+  },
+};
```
Comment (alannnc) [inline apps/web/next.config.js:235]: This shouldn't affect self hosted.

File: apps/web/next.config.js [hunk 2]
```diff
@@ -228,6 +229,18 @@ const nextConfig = {
...
+
+const sentryWebpackPluginOptions = {
+  silent: true, // Suppresses all logs
```
Comment (alannnc) [inline apps/web/next.config.js:238]: Same as above

File: apps/web/next.config.js [hunk 3]
```diff
@@ -228,6 +229,18 @@ const nextConfig = {
...
 };
 
-module.exports = () => plugins.reduce((acc, next) => next(acc), nextConfig);
+const moduleExports = () => plugins.reduce((acc, next) => next(acc), nextConfig);
+
+// Sentry should be the last thing to export to catch everything right
+module.exports = process.env.NEXT_PUBLIC_SENTRY_DSN
+  ? withSentryConfig(moduleExports, sentryWebpackPluginOptions)
```
Comment (alannnc) [inline apps/web/next.config.js:245]: If they don't have any sentry env we should fallback to default export.

**Comment by JeroenReumkens** [comment]
LGTM @alannnc !

**Review by JeroenReumkens — APPROVED** [review]

**Review by hariombalhara — DISMISSED** [review]
Blocking this till crash problem is solved. 
https://threads.com/34432396033?s=hDF7phTfHe6ecRWvnwWRV6
https://www.loom.com/share/f3fa835c5d7f4695be1121f271f8dfaa

I have tried cleaning up node_modules and re-installation but this error persists. I am worried that this might break infra users build.

**Comment by alannnc** [comment]
> LGTM @alannnc !
> 
> Only wondering if the huge update to yarn.lock is a mistake or not?

Thats what I think too just did a clean and fresh install of everything, probably it's not being updated correctly from others and someone has to take it. Help me out and let me know if you generate the same lock.
@JeroenReumkens

**Comment by JeroenReumkens** [comment]
@alannnc  The only thing if you delete the yarn.lock is that lots of packages get upgraded to a new version, so in theory that could result in stuff being broken because of a package update. I my opinion you usually don't want purge a full yarn lock file, and only update one dependency at a time (which will update that dependency in the lock file as well).

**Comment by hariombalhara** [comment]
@JeroenReumkens I did a checkout of yarn.lock from main and run a `yarn update` again. The file has pretty less changes now.

**Comment by hariombalhara** [comment]
Screenshot 2022-11-03 at 4 58 08 PM
Tested a sample error and it's recorded in Sentry.
merging it now.

**Pull request merged**